ClassAttendance/src/WebUI/Controllers/AccountController.cs

```diff
@@ -0,0 +1,111 @@
+﻿using BLL.Exceptions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+using WebUI.Facades;
+using WebUI.Models;
+using WebUI.Models.Account;
+
+namespace WebUI.Controllers
+{
+    public class AccountController : Controller
+    {
+        private ILogger<AccountController> _logger;
+        private AccountFacade _accountFacade;
+
+        public AccountController(ILogger<AccountController> logger, AccountFacade accountFacade)
+        {
+            _logger = logger;
+            _accountFacade = accountFacade;
+        }
+
+        public Uri Referer => new Uri(Request.Headers["Referer"].ToString());
+
+        [AllowAnonymous]
+        [HttpGet]
+        public IActionResult Login()
+        {
+            var model = new LoginModel();
+
+            return View(model);
+        }
+
+        [AllowAnonymous]
+        [HttpPost]
+        public async Task<IActionResult> Login(LoginModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    await _accountFacade.SignInAsync(model);
+                    _logger.LogInformation("User signed in");
+
+                    return RedirectToAction("Index", "Home");
```

Comment:
nameof

Reply:
I'm trying to call other contoller method, so i can't directly call nameof(). I could add something like [this](https://stackoverflow.com/a/36626343/12741272), but i think it is too complex for this.

Reply:
Resolved.